src/styles/main.scss

```diff
@@ -10,5 +10,21 @@ body {
   box-sizing: border-box;
 }
 
+@import "./utilits/variables";
+
 @import "_header.scss";
+
 @import "_catalog.scss";
+
+.catalog {
+  display: grid;
+  grid-template-columns: repeat(4, 200px);
+  gap: $gap;
+  justify-content: center;
+
+  @media (max-width: $card-width*4 + $gap*3 + 80px) {
+    & {
+      grid-template-columns: repeat(auto-fill, $card-width);
+    }
+  }
+}
```

Comment:
You forget to make it specific:
Media queries
There should be four columns starting at 1024px
There should be three columns starting at 768px
There should be two columns starting at 488px
There should be one column below 488px